File: investigation/kotlin_probe/module1/src/kz/greetgo/kotlin_probe/module1/Bean1.java
package kz.greetgo.kotlin_probe.module1;

import kz.greetgo.depinject.core.Bean;
import kz.greetgo.depinject.core.BeanGetter;
import kz.greetgo.kotlin_probe.module2.Bean2;

@Bean
public class Bean1 {

  public BeanGetter<Bean2> bean2;

  public void printHelloWorld() {
    System.out.println("Hello World from " + getClass());
    bean2.get().printHelloWorld();
  }
}

File: investigation/kotlin_probe/module1/src/kz/greetgo/kotlin_probe/module1/Bean1.kt
package kz.greetgo.kotlin_probe.module1

import kz.greetgo.depinject.core.Bean
import kz.greetgo.depinject.core.BeanGetter
import kz.greetgo.kotlin_probe.module2.Bean2

@Bean
class Bean1 {

  lateinit var bean2: BeanGetter<Bean2>

  fun printHelloWorld() {
    println("Hello World from " + javaClass.simpleName + " (Under Kotlin!!!)")
    bean2.get().printHelloWorld()
  }
}
